### ISSUE TITLE
Issue with build menu constructions never starting

### ISSUE THREAD

**Issue by thomp2th** [DFHack/dfhack#3234]
So, basically, some of my constructions are never getting started. They are stuck on "Construction not yet started" and I believe the issue is DFHack because I never ran into it until now. I can build, say, 200 floor tiles, and maybe 10 or so will never start construction while the rest get built. Unsuspending does nothing, they don't get suspended or cancelled.  Attached is the world save file and the only mod I use besides DFHack (see thru smoothing designations).  There is a mining guild hall in the first cavern layer that has 2 floors experiencing this issue.
[current.zip](https://github.com/DFHack/dfhack/files/11239921/current.zip)
[2898549749 (2).zip](https://github.com/DFHack/dfhack/files/11239922/2898549749.2.zip)

**Comment by myk002** [comment]
Thanks! I'll look into it.

**Comment by thomp2th** [comment]
Apologies, I uploaded the wrong save - here's the correct one.  Too big to upload:  https://drive.google.com/file/d/1-1kuiTu_TB7RviFmmsm-7GbLv7oP6g5R/view?usp=share_link

**Comment by myk002** [comment]
Where are the relevant constructions? I see a few suspended ones, and one that's inaccessible due to a missing floor, but this ins't what you're talking about, right?
![image](https://user-images.githubusercontent.com/977482/232256575-82aa8caa-4ea8-436b-b6e1-358aee76aa30.png)

**Comment by myk002** [comment]
n/m you said the floors in the guildhall are the problematic ones. checking them out now.

**Comment by myk002** [comment]
Ok, I tracked down the problem. The item attached to that bit of flooring became inaccessible between the time that `buildingplan` attached it to the job and the time that the job was picked up for completion:
![image](https://user-images.githubusercontent.com/977482/232257875-ca3d391c-9c2b-4261-9c8e-1b7123b5eb0d.png)

Note that this can happen in vanilla as well. This is not something `buidlingplan` can easily fix by itself, since the job only became problematic *after* buildingplan attached the item and released it to vanilla mechanics. We *could* create some sort of scanner that checks the accessibility of already-attached items, but the original `buildingplan` filter used to plan the building in the first place may no longer be accessible (if the filter has been changed since the building was originally planned).

When this happens in vanilla, you usually have no recourse but to cancel the building and try again. This is perhaps an area where DFHack could improve the experience, but I don't think this is a bug, per se.

**Comment by myk002** [comment]
Here's an idea of something we could do:

We could add an overlay that appears when looking at an inactive building job, showing details for why the job is inactive; include a button to zoom to the attached item (maybe it's behind a gate and all the player has to do is open the gate)

**Comment by thomp2th** [comment]
Yeah, that would be good!  Or maybe is there any way to restart unstartable
jobs when you do suspend unreachable i wonder?

**Comment by myk002** [comment]
> Or maybe is there any way to restart unstartable jobs when you do suspend unreachable i wonder?

That's possible too. The code would have to ask the player what item to use instead, though, or the building could be canceled and re-planned with the current `buildingplan` filters.
